Bottom: 5188c490b5cf38c9cc37c248447020c853cf6a75
Top:    8b020b7d96584f8ae0846b1cdba14c2d55b75892
Author: Ben Greear <greearb@candelatech.com>
Date:   2013-05-09 11:56:19 -0700

vlan: Support sending custom Ethernet CRC.

If the underlying device supports sending custom Ethernet CRC,
then enable that feature in the VLANs as well.

Signed-off-by: Ben Greear <greearb@candelatech.com>


---

diff --git a/net/8021q/vlan_dev.c b/net/8021q/vlan_dev.c
index 1b0d28e..1a0891e 100644
--- a/net/8021q/vlan_dev.c
+++ b/net/8021q/vlan_dev.c
@@ -581,6 +581,9 @@ static int vlan_dev_init(struct net_device *dev)
 
 	dev->vlan_features = real_dev->vlan_features & ~NETIF_F_ALL_FCOE;
 
+	if (netif_supports_nofcs(real_dev))
+		dev->priv_flags |= IFF_SUPP_NOFCS;
+
 	/* ipv6 shared card related stuff */
 	dev->dev_id = real_dev->dev_id;
